src/units.rs

```diff
@@ -551,6 +588,50 @@ fn actual_multiply(left: Number, right: Number, swapped: bool) -> Result<Number,
     };
     let kilometers = Number::new(result, Kilometer);
     Ok(convert(kilometers, final_unit)?)
+  } else if lcat == DigitalDataRate && rcat == Time {
+    // 8 megabytes per second * 1 minute
+    let data_rate_value = left.value * left.unit.weight();
+    let seconds = convert(right, Second)?;
+    let result = data_rate_value * seconds.value;
+    let final_unit = match left.unit {
+      BitsPerSecond => Bit,
+      KilobitsPerSecond => Kilobit,
+      MegabitsPerSecond => Megabit,
+      GigabitsPerSecond => Gigabit,
+      TerabitsPerSecond => Terabit,
+      PetabitsPerSecond => Petabit,
+      ExabitsPerSecond => Exabit,
+      ZettabitsPerSecond => Zettabit,
+      YottabitsPerSecond => Yottabit,
+      KibibitsPerSecond => Kibibit,
+      MebibitsPerSecond => Mebibit,
+      GibibitsPerSecond => Gibibit,
+      TebibitsPerSecond => Tebibit,
+      PebibitsPerSecond => Pebibit,
+      ExbibitsPerSecond => Exbibit,
+      ZebibitsPerSecond => Zebibit,
+      YobibitsPerSecond => Yobibit,
+      BytesPerSecond => Byte,
+      KilobytesPerSecond => Kilobyte,
+      MegabytesPerSecond => Megabyte,
+      GigabytesPerSecond => Gigabyte,
+      TerabytesPerSecond => Terabyte,
+      PetabytesPerSecond => Petabyte,
+      ExabytesPerSecond => Exabyte,
+      ZettabytesPerSecond => Zettabyte,
+      YottabytesPerSecond => Yottabyte,
+      KibibytesPerSecond => Kibibyte,
+      MebibytesPerSecond => Mebibyte,
+      GibibytesPerSecond => Gibibyte,
+      TebibytesPerSecond => Tebibyte,
+      PebibytesPerSecond => Pebibyte,
+      ExbibytesPerSecond => Exbibyte,
+      ZebibytesPerSecond => Zebibyte,
+      YobibytesPerSecond => Yobibyte,
+      _ => Bit,
+    };
```

Comment:
Maybe we could have an `adjust_unit()` function here that works similarly to `to_ideal_unit()`, but keeps the unit variations. If we divide by a -bit unit, we return a -bit unit etc:
```
-bit => -bit
-byte => -byte
-bibit => -bibit
-bibyte => -bibyte
```

Reply:
Could you please expand a bit on what you're envisaging?

Reply:
With other units, we usually return units based just on how big the value is. For example, a value equal to 1km always returns `Kilometer`.

In this case though, it might make more sense to do this:
`10 byte/s * 100s = 1 Kilobyte`
`80 bit/s * 100s = 8 Kilobit`

So when the unit we start with ends in `-byte`, we return a unit that ends in `-byte`. Pseudocode:
```rust
fn digital_storage_variant(unit) {
  match unit {
    bit | kilobit | ... => DigitalStorageVariant::Bit,
    byte | kilobyte | ... => DigitalStorageVariant::Byte,
    kibibit | mebibit | ... => DigitalStorageVariant::Bibit,
    kibibyte | mebibyte | ... => DigitalStorageVariant::Bibyte,
  }
}

fn adjust_unit(number) {
  let value = number.value * number.unit.weight();
  match digital_storage_variant(number.unit) {
    DigitalStorageVariant::Bit => {
      if value >= 1000000000000000000000000 {
        Number::new(value/Year.weight(), Yottabit);
      } else if value >= 1000000000000000000000 {
      // ...
    }
    // ...
  }
}
```

Reply:
I've gotta be honest, I don't fully understand the change you're proposing. I've updated the PR with the name change, but I think it would be better if you implemented the `adjust_unit` function you're envisaging on top of this work.

Reply:
To explain less technically, this is how the PR works now:
`80 bit/s * 500 hour = 144000000 Bit`
`80 byte/s * 500 hour = 144000000 Bit`
`80 kibibyte/s * 500 hour = 144000000 Kibibyte`

And this is what I suggested:
`80 bit/s * 500 hour = 144 Megabit`
`80 byte/s * 500 hour = 144 Megabyte`
`80 kibibyte/s * 500 hour = 137.3291015625 Gibibyte`

Would be fine to leave that for later though